### PR TITLE
Fix all compilation warnings in go-guru.el

### PR DESCRIPTION
**Pull request by phst**

### CODE REVIEW

File: go-guru.el [hunk 1]
```diff
@@ -117,7 +117,7 @@
   (define-key m ">" #'go-guru-callees)
   (define-key m "x" #'go-guru-expand-region)) ;; x for expand
 
-(define-key go-mode-map (kbd "C-c C-o") #'go-guru-map)
+(define-key go-mode-map (kbd "C-c C-o") 'go-guru-map)
```
Comment (dominikh) [inline go-guru.el]: What's the warning that triggered this change?

File: go-guru.el [hunk 2]
```diff
@@ -450,6 +453,22 @@ highlights from previously highlighted identifier."
   (go-guru-unhighlight-identifiers)
   (go-guru--hl-identifier))
 
+;;;###autoload
```
Comment (dominikh) [inline go-guru.el]: Why was this chunk of code moved around?

File: go-guru.el (hunk 1 above)
Comment (phst) [inline go-guru.el]: "Function might not be defined". Indeed `go-guru-map` isn't a real function that could be called, but a keymap.

Comment (dominikh) [inline go-guru.el]: Right, of course.

**Pull request closed without merging**